framework/web/Request.php

```diff
@@ -535,15 +579,11 @@ public function getHostInfo()
     {
         if ($this->_hostInfo === null) {
             $secure = $this->getIsSecureConnection();
+            $domain = $this->getDomain();
             $http = $secure ? 'https' : 'http';
-            if (isset($_SERVER['HTTP_HOST'])) {
-                $this->_hostInfo = $http . '://' . $_SERVER['HTTP_HOST'];
-            } elseif (isset($_SERVER['SERVER_NAME'])) {
-                $this->_hostInfo = $http . '://' . $_SERVER['SERVER_NAME'];
+            if($domain !== null) {
                 $port = $secure ? $this->getSecurePort() : $this->getPort();
-                if (($port !== 80 && !$secure) || ($port !== 443 && $secure)) {
-                    $this->_hostInfo .= ':' . $port;
-                }
+                $this->_hostInfo = $http . '://' . $domain . ':' . $port;
```

Comment:
I think it changes behavior i.e. `:80` and `:433` will be added while previously it was not.